sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -562,13 +577,27 @@ case class IntegralDivide(
       case d: DecimalType =>
         d.asIntegral.asInstanceOf[Integral[Any]]
     }
-    (x, y) => {
-      val res = integral.quot(x, y)
-      if (res == null) {
-        null
-      } else {
-        integral.asInstanceOf[Integral[Any]].toLong(res)
+    val _div =
+      (x: Any, y: Any) => {
+        val res = integral.quot(x, y)
+        if (res == null) {
+          null
+        } else {
+          integral.asInstanceOf[Integral[Any]].toLong(res)
+        }
       }
+
+    dataType match {
+      case LongType if failOnError =>
```

Comment:
This is not necessary because this class has `override def dataType: DataType = LongType`

Reply:
I think we need to check `child.dataType`

Reply:
Yeah you are right